### PR TITLE
Process docstring cli testing

### PR DESCRIPTION
**Pull request by jnikula**

### CODE REVIEW

File: src/hawkmoth/__main__.py [hunk 1]
```diff
@@ -45,11 +57,20 @@ def main():
...
+                        ],
+                        help='Process docstring.')
+    compat.add_argument('--compat',
+                        choices=[
+                            'none',
+                            'javadoc-basic',
+                            'javadoc-liberal',
+                            'kernel-doc'
+                        ],
                         help='Compatibility options. See cautodoc_compat.')
```
Comment (BrunoMSantos) [inline src/hawkmoth/__main__.py:74]: Consider raising `DeprecationWarning` :man_shrugging:

**Review by BrunoMSantos — APPROVED** [review]
Looks good ;)

File: src/hawkmoth/__main__.py (hunk 1 above)
Comment (BrunoMSantos) [reply src/hawkmoth/__main__.py:74]: Side note: we may also want to institutionalize a deprecation policy so we have a regular, well advertised schedule for dropping things. We have an important user now and everything <.<

Comment (jnikula) [reply src/hawkmoth/__main__.py:74]: Yeah, the deprecation has been haphazard so far. If it's release based, I guess it'll have to be major releases. And we're not even at 1.0. :grin:

**Pull request merged**